spec/several_assignments_spec.rb

```diff
@@ -1,10 +1,12 @@
+# frozen_string_literal: true
+
 describe "attribute with several assignments" do
   before do
     class Test::Foo
       extend Dry::Initializer
 
-      option :bar, proc(&:to_s),    optional: true
-      option :"some foo", as: :bar, optional: true
+      option :bar, proc(&:to_s), optional: true
+      option :some_foo, as: :bar, optional: true
```

Comment:
String keyword argument names aren't allowed in 2.6. This is perhaps a very small breaking change but frankly its unlikely anyone on 2.6 was taking advantage of it.

Reply:
I think here is a potential for the backward incompatibility. This test is about not string, but symbol keys. What is essential is that the keys don't need to be a valid ruby methods.

For example, we should can handle this assignment:

```ruby
class Foo
  extend Dry::Initializer
  
  option :"!!! my weird key", as: :weird_key
end

foo = Foo.new("!!! my weird key": 1) # notice the key is a symbol, not a string
foo.weird_key # => 1
```

The goal of this feature to make it possible handling any keys in the income hash (maybe with a forced symbolization like below)

```ruby
data = { 1 => 2 }.transform_keys { |k| k.to_s.to_sym }
```